src/binding-hash.h

```diff
@@ -18,34 +18,38 @@
 #define WABT_BINDING_HASH_H_
 
 #include "common.h"
-#include "vector.h"
+
+#include <string>
+#include <unordered_map>
 
 namespace wabt {
 
 struct Binding {
+  explicit Binding(int index) : index(index) {
+    WABT_ZERO_MEMORY(loc);
+  }
+  Binding(const Location& loc, int index) : loc(loc), index(index) {}
+
   Location loc;
-  StringSlice name;
   int index;
 };
 
-struct BindingHashEntry {
-  Binding binding;
-  struct BindingHashEntry* next;
-  struct BindingHashEntry* prev; /* only valid when this entry is unused */
-};
-WABT_DEFINE_VECTOR(binding_hash_entry, BindingHashEntry);
-
-struct BindingHash {
-  BindingHashEntryVector entries;
-  BindingHashEntry* free_head;
-};
-
-Binding* insert_binding(BindingHash*, const StringSlice*);
-void remove_binding(BindingHash*, const StringSlice*);
-bool hash_entry_is_free(const BindingHashEntry*);
-/* returns -1 if the name is not in the hash */
-int find_binding_index_by_name(const BindingHash*, const StringSlice* name);
-void destroy_binding_hash(BindingHash*);
+typedef std::unordered_multimap<std::string, Binding> BindingHash;
```

Comment:
Somehow, this code seems to be missing a class that cleanly captures the API  better than this.